### DISCUSSION TITLE
Visually differentiate localized fields from non-localized fields

### DISCUSSION THREAD

**Discussion by zubricks** [payloadcms/payload#4200] [Roadmap]
It's tough to identify at a glance which fields have `localized: true,` and which fields do not. We should consider extending the built-in field types with a way to denote if they are localized or not e.g. When editing a single locale, show the current locale in a pill next to the field label to show that it is localized.

EDIT: **RELEASED IN [3.0](https://github.com/payloadcms/payload/releases/tag/v3.0.0)**

**Comment by tylandavis** [comment]
### Design update 11/29/23

To show which fields are localized, a pill with the current locale will be added next to the label of the field:

![localized-fields-single](https://github.com/payloadcms/payload/assets/89618855/e0fe1968-7804-40df-8824-7904cb6762bf)

**Reply by nexflo** [reply]
Is this commited to code? 
Seems like a tiny but massive change for localisastion?

**Reply by benjick** [reply]
Looks great, any way to have this today?

**Reply by tylandavis** [reply]
You could achieve this yourself by replacing the label with a custom component, and use a combination of Payload [React hooks](https://payloadcms.com/docs/admin/hooks) to determine which fields are localized and which locale is currently selected.

**Reply by benjick** [reply]
I ended up creating a plugin which traverses every field and checks if localized = true and wraps the underlying component so I can add a border

**Reply by nexflo** [reply]
I just saw this also on screenshots on the official payload site... so i thoughts its flying around as a plugin somewhere...or as a native feature. 
@benjick you thinking about releasing the plug? :)

**Reply by benjick** [reply]
@nexflo I could if you want, but if you're on 3.0 there's already https://github.com/r1tsuu/payload-enchants/tree/master/packages/better-localized-fields which I based my 2.x plugin on 

(moved this comment into the correct thread)

**Reply by Because789** [reply]
Any news on a native implementation of this? I get that you are pushing for v3 stable now, but since I really need a visual cue for localized fields, a status update would be nice before I start implementing something myself.

Btw. https://github.com/r1tsuu/payload-enchants/tree/master/packages/better-localized-fields looks promising, but it doesn't work with the current beta.

**Reply by rolandsaven** [reply]
Seems like a version of this had been pushed to beta yesterday https://github.com/payloadcms/payload/pull/8602

**Comment by tylandavis** [comment]
🚀 This is included in version [v3.0.0-beta.118](https://github.com/payloadcms/payload/releases/tag/v3.0.0-beta.118)

**Reply by nexflo** [reply]
Sweet! Now in 3.0.0
The interface sketched above clearly distinguishes language and title quite a bit more then the current implementation tho.

**Reply by PP-Tom** [reply]
Agreed, though I wonder if the decision was made because it looked too much like a button as it's styled much like the buttons when selecting a document in a collection view.

**Reply by carlcs** [reply]
@tylandavis it does not show on checkbox fields.

**Reply by carlcs** [reply]
And I agree with above comments, would be nice if it didn’t look like part of the field name. I like they way Craft CMS just uses an icon. You already have the currently selected locale name in the locale selector.
 
<img width="310" alt="Bildschirmfoto 2024-12-13 um 12 36 09" src="https://github.com/user-attachments/assets/70ad4636-33b0-4159-9565-80931256bbce" />
